### PR TITLE
Add project created_at support and project list sorting

### PR DESCRIPTION
**Pull request by masakiaota**
## Summary
- add project `created_at` support with legacy DB backfill
- add Project List sort controls for created/name/documents/pending documents
- align backend schema/docs examples with the new project field

## Testing
- uv run pytest tests/test_projects.py tests/test_import_export.py
- npm test -- --run src/test/projectsPage.test.tsx src/test/projectShellSettings.test.tsx src/test/useProjectBundle.test.ts

### CODE REVIEW

File: backend/src/layered_span_studio_backend/repositories/projects.py [hunk 1]
```diff
@@ -37,6 +37,31 @@ def _parse_timestamp(value: Any) -> Optional[float]:
...
+    columns = {row["name"] for row in conn.exec_driver_sql("PRAGMA table_info(project)").mappings()}
+    if "created_at" in columns:
+        return
+    created_at = _project_db_timestamp(db_path)
+    conn.exec_driver_sql("ALTER TABLE project ADD COLUMN created_at TEXT")
+    conn.execute(
+        project_table.update()
+        .where(project_table.c.created_at.is_(None))
+        .values(created_at=created_at)
+    )
```
[the remark this answers is not in the document]
Comment (masakiaota) [reply backend/src/layered_span_studio_backend/repositories/projects.py]: Addressed. I agreed this was a real race/null-backfill gap. `_ensure_project_created_at_column` now always backfills `created_at IS NULL` rows even when the column already exists, and it tolerates a duplicate-column race during `ALTER TABLE`. I also added a regression test that covers the case where the column exists but the row value is still NULL.

File: backend/src/layered_span_studio_backend/repositories/projects.py [hunk 2]
```diff
@@ -37,6 +38,41 @@ def _parse_timestamp(value: Any) -> Optional[float]:
...
+    columns = {row["name"] for row in conn.exec_driver_sql("PRAGMA table_info(project)").mappings()}
+    created_at = _project_db_timestamp(db_path)
+    should_backfill = False
+    if "created_at" not in columns:
+        should_backfill = True
+        try:
+            conn.exec_driver_sql("ALTER TABLE project ADD COLUMN created_at TEXT")
+        except OperationalError as exc:
+            if "duplicate column name" not in str(exc).lower():
+                raise
```
[the remark this answers is not in the document]
Comment (masakiaota) [reply backend/src/layered_span_studio_backend/repositories/projects.py]: Addressed. I agreed this intermittent lock risk was real on the first-read migration path. I kept the migration local to the read path, but added a short retry/backoff for SQLITE_BUSY / locked errors around the schema/backfill step so concurrent access to the same legacy project DB is less likely to surface as a transient 500.

File: frontend/src/test/projectsPage.test.tsx [hunk 3]
```diff
@@ -43,6 +44,31 @@ function createImportFile(payload: unknown) {
...
+    summary,
+  };
+}
+
+function getRenderedProjectNames() {
+  return Array.from(document.querySelectorAll("h6"))
+    .map((node) => node.textContent ?? "")
+    .filter((text) => text !== "Layered Span Studio");
+}
+
```
[the remark this answers is not in the document]
Comment (masakiaota) [reply frontend/src/test/projectsPage.test.tsx]: Addressed. I agreed the old helper was too brittle. Instead of scraping all `h6` elements and filtering a hard-coded header string, the test now reads project names via a dedicated `data-testid` on the project title element. That keeps the assertion scoped to the actual project cards and avoids accidental breakage from unrelated headings.
